src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -158,12 +160,12 @@ public AppStatus status(String id) {
 		return Flux.fromStream(request.getDefinition().getProperties().entrySet().stream());
 	}
 
-	/**
-	 * TODO: Should we join properties with override, or replace?
-	 *
-	 * @param request
-	 * @return
-	 */
+	private String deploymentId(AppDeploymentRequest request) {
+		return String.format("%s-%s",
+				request.getEnvironmentProperties().get(GROUP_PROPERTY_KEY),
```

Comment:
That sounds reasonable, I'll create an issue for that when this is closed and also add the `IllegalArgumentException` that Mark mentioned.